taskcluster/async/asyncclient.py

```diff
@@ -138,14 +149,16 @@ class AsyncBaseClient(BaseClient):
                     superExc=rerr
                 )
 
-            # Handle non 2xx status code and retry if possible
-            try:
-                response.raise_for_status()
-                if response.status == 204:
-                    return None
+            status = response.status
+            if status == 204:
+                return None
 
-            except aiohttp.HttpProcessingError as rerr:
-                status = response.status
+            # Handle non 2xx status code and retry if possible
+            if status < 200 or status >= 300:
+                rerr = aiohttp.ClientResponseError(
```

Comment:
As I mentioned before, I don't think there's any reason for us to be throwing an aiohttp internal exception here.  Let's just throw the `exceptions.TaskclusterRestFailure` with the properties here.